AmmadAmjad-CodeTest/Server/models/user.js

```diff
@@ -0,0 +1,138 @@
+var bcrypt = require('bcrypt');
+var _ = require('underscore');
+var cryptojs = require('crypto-js');
+var jwt = require('jsonwebtoken');
+
+module.exports = function(sequelize, DataTypes) {
+	var user = sequelize.define('user', {
+		email: {
+			type: DataTypes.STRING,
+			allowNull: false,
+			unique: true,
+			validate: {
+				isEmail: true
+			}
+		},
+		salt: {
+			type: DataTypes.STRING
+		},
+		password_hash: {
+			type: DataTypes.STRING
+		},
+		name: {
+			type: DataTypes.STRING,
+			allowNull: false,
+		},
+		password: {
+			type: DataTypes.VIRTUAL,
+			allowNull: false,
+			validate: {
+				len: [7, 100]
+			},
+			set: function(value) {
+				var salt = bcrypt.genSaltSync(10);
+				var hashedPassword = bcrypt.hashSync(value, salt);
+
+				this.setDataValue('password', value);
+				this.setDataValue('salt', salt);
+				this.setDataValue('password_hash', hashedPassword);
+			}
+		},
+		token: {
+			type: DataTypes.VIRTUAL,
+			allowNull: true,
+			set: function(value) {
+				var hash = cryptojs.MD5(value).toString();
+				this.setDataValue('token', value);
+				this.setDataValue('tokenHash', hash);
+			}
+		},
+		tokenHash: {
+			type: DataTypes.STRING,
+			allowNull: true
+		}
+	}, {
+		hooks: {
+			beforeValidate: function(user, options) {
+				// user.email
+				if (typeof user.email === 'string') {
+					user.email = user.email.toLowerCase();
+				}
+			}
+		}
+	});
+
+
+	user.authenticate = function(body) {
+		return new Promise(function(resolve, reject) {
```

Comment:
I see some files are using spaces and some files are using tabs, it would be good to make them consistent.